sql/expression/function/space.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"github.com/shopspring/decimal"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// Space implements the sql function "space" which returns a string with the number of spaces specified by the argument
+type Space struct {
+	*UnaryFunc
+}
+
+var _ sql.FunctionExpression = (*Space)(nil)
+var _ sql.CollationCoercible = (*Space)(nil)
+
+func NewSpace(arg sql.Expression) sql.Expression {
+	return &Space{NewUnaryFunc(arg, "SPACE", types.LongText)}
+}
+
+// Description implements sql.FunctionExpression
+func (s *Space) Description() string {
+	return "return a string of the specified number of spaces."
+}
+
+// CollationCoercibility implements the interface sql.CollationCoercible.
+func (s *Space) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID, coercibility byte) {
+	return sql.Collation_binary, 5
+}
+
+// Eval implements the sql.Expression interface
+func (s *Space) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	val, err := s.EvalChild(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if val == nil {
+		return nil, nil
+	}
+
+	// TODO: better truncate integer handling
+	var v interface{}
+	switch val := val.(type) {
+	case float32:
+		v = int64(val)
+	case float64:
+		v = int64(val)
+	case decimal.Decimal:
+		v = val.IntPart()
```

Comment:
Looks like MySQL will round float and decimal values to the nearest integer, instead of just discarding the fractional component. So, for example, a decimal or float of `1.99` passed in should result in 2 spaces, instead of 1. (floats inside a literal string **do** get simply truncated though, like you've got implemented below.)

```
select concat("'", space(CAST('1.999' AS DECIMAL)), "'");
+---------------------------------------------------+
| concat("'", space(CAST('1.999' AS DECIMAL)), "'") |
+---------------------------------------------------+
| '  '                                              |
+---------------------------------------------------+
```